spec/rspec/core/formatters/exception_presenter_spec.rb

```diff
@@ -12,6 +12,10 @@ module RSpec::Core
       allow(example.execution_result).to receive(:exception) { exception }
       example.metadata[:absolute_file_path] = __FILE__
       allow(exception).to receive(:cause) if RSpec::Support::RubyFeatures.supports_exception_cause?
+      allow(Exception).to receive(:===).and_wrap_original do |original, *other|
+        return false if String === true
```

Comment:
I'm having a hard time understanding how `String === true` works and when it can evaluate to `true`.

Reply:
This doesn't work, it messes with the exceptions to much, but the premiss was this, we're detecting if `cause` is returning exceptions, `Exception === other` will return true for any instance that extends it, the deliberately broken example used a string so I was saying, hey all things are exceptions except for strings.